util/include/rusefi/rusefi_time_types.h

```diff
@@ -54,5 +59,23 @@ efitick_t getTimeNowNt();
 #define US_PER_SECOND_F 1000000.0
 #define US_PER_SECOND_LL 1000000LL
 
-#define MS2US(MS_TIME) ((MS_TIME) * 1000)
-#define US2MS(US_TIME) ((US_TIME) / 1000)
+template<typename arithmetic_type> constexpr efitimeus_t USOF(const arithmetic_type& us) {
```

Comment:
I would avoid templating the conversion functions. There are scenarios where we want/need to avoid a 64b->float or 64b divide (both are slow as there's no hardware for it).

Reply:
Functions USOF and USOFMS are actually used only in mock and unit tests. I'm not sure that we should worry about perfomance there. Maybe this header is not good place for such  the other hand I've no idea where is appropriate place for such functions. But I'm not sure what place is more appropriate place for functions to share between `lib-time-mocks.cpp` and unit tests. Do you have any suggestion?

Reply:
@kifir23917 new header something with obviously test name?

Reply:
I've got rid of template functions and moved conversions float -> int64_t to tests